### PR TITLE
[Issue #34] - removing more instances of .unwrap() from documentation in favor of try!()

### PR DESCRIPTION
**Pull request by hiimtaylorjones**
As per #34, there were still a few instances of .unwrap() left in documentation. So, this (hopefully) covers some more ground towards eliminating these instances. 

Please note that `try!()` is a new Rust idiom to me personally, but I think I was able to get the hang of it pretty quickly. There's definitely a chance I messed something up though!

### CODE REVIEW

File: diesel/src/expression/expression_methods/escape_expression_methods.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use types::VarChar;
 /// #     use diesel::insert;
 /// #     let connection = establish_connection();
 /// #     insert(&NewUser { name: "Ha%%0r".into() }).into(users)
-/// #         .execute(&connection).unwrap();
+/// #         .try!(execute(&connection));
```
Comment (Ryman) [inline diesel/src/expression/expression_methods/escape_expression_methods.rs]: Sadly, macros (which `try!` is) can't be called like methods, so the entire expression needs to be wrapped. You can try...
```rust
try!(insert(..).into(..).execute(..));
```

File: diesel/src/expression/expression_methods/global_expression_methods.rs [hunk 2]
```diff
@@ -97,8 +97,8 @@ pub trait ExpressionMethods: Expression + Sized {
     /// # fn main() {
     /// #     use self::users::dsl::*;
     /// #     let connection = establish_connection();
-    /// #     connection.execute("INSERT INTO users (name) VALUES
-    /// #         ('Jim')").unwrap();
+    /// #     try!(connection.execute("INSERT INTO users (name) VALUES
```
Comment (Ryman) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Slightly different problem with these, as `main` doesn't expect to return a `Result`, so the `try!` will fail. Not sure if there's any benefit to switching to `try!` here as this line is hidden from the output (It's prefixed with '#')

File: diesel/src/expression/expression_methods/escape_expression_methods.rs (hunk 1 above)
Comment (hiimtaylorjones) [inline diesel/src/expression/expression_methods/escape_expression_methods.rs]: 😢 Well that's a bummer. 

Thanks for the pointer! Pushing up the change in a few.

File: diesel/src/expression/expression_methods/global_expression_methods.rs (hunk 2 above)
Comment (hiimtaylorjones) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Hmmmmmm.

Maybe we should leave unwrap() in there then. Whatcha think @sgrif?

Comment (sgrif) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: We shouldn't remove it anywhere except documentation. This line isn't displayed

Comment (hiimtaylorjones) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Oh! I see how the docs are being picked up and interpreted now. Will do.

**Comment by sgrif** [comment]
I don't think any of these cases really need to be changed. We can't use `try!` for the most part in the main functions, what we're more looking for is anywhere that we're doing `assert_eq!(foo, bar.unwrap())` where we could instead make a change like `assert_eq!(Ok(foo), bar)`. I'm not sure if there are any of those cases left or not.

**Pull request closed without merging**